porcupine/textutils.py

```diff
@@ -668,7 +668,7 @@ def dedent(self, location: str) -> bool:
                 # not deleting from start of line, delete previous char instead
                 location = f"{location} - 1 char"
 
-            if self.get(location).isspace():
+            if self.get(location).isspace() and self.get(location) != "\n":
```

Comment:
```suggestion
            # Deleting spaces this way is annoying. But that is somewhat
            # intentional because mixing tabs and spaces is bad.
            if self.get(location) in {" ", "\t"}:
```